src/modules/solution/dtos/create-solution-request-body.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { PartialType, PickType } from "@nestjs/swagger";
+import { Lesson, LessonSolution } from "@prisma/client";
+import { IsRecord } from "@src/decorators/is-record.decorator";
+import { IsDataURI, IsInt, IsNotEmpty, IsNumber, IsOptional, IsString, IsUrl, Min, MinLength } from "class-validator";
+import { SolutionEntity } from "../entities/solution.entity";
+
+export class CreateSolutionRequestBodyDto extends PickType(SolutionEntity, [
+  'lessonId',
+  'description',
+  'relatedLink'
+]) {
+  @IsRecord<Lesson>({ model: 'lesson', field: 'id' }, false)
+  @IsInt()
+  @Min(1)
+  @IsNotEmpty()
+  lessonId: number;
+
+  @IsString()
+  @IsNotEmpty()
+  description: string;
+
+  @IsUrl({ require_protocol: true, require_valid_protocol: true })
+  @IsString()
+  @IsOptional()
+  relatedLink: string;
```

Comment:
IsOptional 데코레이터가 붙어있으면 옵셔널 프로퍼티이기 때문에 처럼 optional 한 타입으로 가져가야할거같아요
```suggestion
  relatedLink?: string;
```